File: ext/capacitor-native-webauthn/android/src/main/java/com/darkedges/capacitor/webauthn/WebAuthn.java
package com.darkedges.capacitor.webauthn;

import android.util.Log;
import androidx.credentials.CredentialManager;

public class WebAuthn {
    private CredentialManager credentialManager;

    public boolean isWebAuthnAvailable() {
        return isAvailable(WebAuthnTypes.WEBAUTHN);
    }
    public boolean isWebAuthnAutoFillAvailable() {
        return isAvailable(WebAuthnTypes.WEBAUTHNAUTOFILL);
    }

    private boolean isAvailable(WebAuthnTypes webAuthnType) {
        boolean val = false;
        switch(webAuthnType) {
            case WEBAUTHN:
                val = false;
                break;
            case WEBAUTHNAUTOFILL:
                val=false;
                break;
        }
        return val;
    }

    public void setCredentialManager(CredentialManager credentialManager) {
        Log.i("setCredentialManager", String.valueOf(credentialManager));
        this.credentialManager=credentialManager;
    }

    public CredentialManager getCredentialManager() {
        return this.credentialManager;
    }
}

File: ext/capacitor-native-webauthn/ios/Plugin/WebAuthn.swift
import Foundation

@objc public class WebAuthn: NSObject {
    private var credentialManager: String = ""
    
    @objc public func isWebAuthnAvailable() -> Bool {
        return isAvailable(WebAuthnTypes.WEBAUTHN)
    }
    
    @objc public func isWebAuthnAutoFillAvailable() -> Bool {
        return isAvailable(WebAuthnTypes.WEBAUTHNAUTOFILL);
    }
    
    @objc public func setCredentialManager(_ credentialManager: String) {
        self.credentialManager = credentialManager
    }
    
    @objc public func getCredentialManager() -> String {
        return self.credentialManager
    }
    
    private func isAvailable(_ webAuthnType: WebAuthnTypes) -> Bool {
        var val:Bool = false
        switch webAuthnType {
        case .WEBAUTHN:
            val = false
        case .WEBAUTHNAUTOFILL:
            val = false
        }
        return val
    }
}
